File: app/rocket.rb
class Rocket
  include AASM

  aasm do
    state :ground, initial: true
    event :launch do
      transitions from: :ground, to: :flight
    end

    state :descent
    event :appogee do
      transitions from: :flight, to: :decent
    end

    state :descent_drogue
    event :deploy_drogue do
      transitions from: :descent, to: :recovery_one
    end

    state :descent_main
    event :deploy_main do
      transitions from: :recovery_one, to: :recovery_two, before: :eject_parachute
    end

    event :touchdown do
      transitions from: [:descent_drogue, :descent_main], to: :ground,
    end

    state :idle
    event :sleep do
      transitions from: :ground, to: :power_saver,
    end
  end
end

File: app/rocket.py
from transitions import Machine

class Rocket(object):
    states = ['asleep', 'ground', 'powered', 'coast', ' descent', 'descent_drogue', 'descent_main']
    transitions = [
        { 'trigger': 'launch', 'source': 'ground', 'dest': 'powered' },
        { 'trigger': 'burnout', 'source': 'powered', 'dest': 'coast' },
        { 'trigger': 'appogee', 'source': 'coast', 'dest': 'descent' },
        { 'trigger': 'deploy_drogue', 'source': 'descent', 'dest': 'descent_drogue' },
        { 'trigger': 'deploy_main', 'source': 'descent_drogue', 'dest': 'descent_main' },
        { 'trigger': 'touchdown', 'source': 'descent_main', 'dest': 'ground' }
        { 'trigger': 'sleep', 'source': 'ground', 'dest': 'asleep' }
        { 'trigger': 'wake', 'source': 'asleep', 'dest': 'ground' }
    ]

    def __init__(self):
        self.state_machine = Machine(model=self, states=Rocket.states, transitions=Rocket.transitions, initial='ground')
